plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java

```diff
@@ -216,7 +218,10 @@ private <T> T sendHttpGetToControllerJson(String path, JsonCodec<T> codec)
     {
         ImmutableMultimap.Builder<String, String> additionalHeadersBuilder = ImmutableMultimap.builder();
         controllerAuthenticationProvider.getAuthenticationToken().ifPresent(token -> additionalHeadersBuilder.put(AUTHORIZATION, token));
-        URI controllerPathUri = uriBuilderFrom(getControllerUrl()).appendPath(path).scheme(scheme).build();
+        HttpUriBuilder httpUriBuilder = (proxyRestUri != null) ?
+                uriBuilderFrom(URI.create(proxyRestUri)) :
```

Comment:
It seems the proxy makes the controller url to not be used.

So only one of them should be provided in configuration. Verify this using an assertion like in `MySqlJdbcConfig`.

Reply:
Can the REST and GRPC proxies co-exist? Or only one of them needs to be set?

Reply:
Yes, when rest proxy is set, controller URL is not used.

Reply:
> Can the REST and GRPC proxies co-exist? Or only one of them needs to be set?

Yes, both are used. when the proxy is enabled:
- all the controller/broker requests are using rest endpoint
- all the server queries using grpc query endpoint.